pkg/internal/infraolly/process/snapshot.go

```diff
@@ -153,19 +155,24 @@ func (pw *linuxProcess) Username() (string, error) {
 		// try to get it from gopsutil and return it if ok
 		pw.user, err = pw.process.Username()
 		if err == nil {
+			if pw.user == "" {
```

Comment:
If the user was empty (I saw this on my machine for some process?) we'd never cache anything and we'd keep on reading the file each time.